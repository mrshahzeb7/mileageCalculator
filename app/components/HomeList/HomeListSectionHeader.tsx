import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {borderRadius, colors, iconSizes, spacing} from '../../theme';
import {Icon} from '../Icon/Icon';

interface Props {
  title: any;
  data: any;
  icon: any;
}

export const HomeListSectionHeader = ({title, icon}: Props) => {
  return (
    <View style={styles.container}>
      <Icon size={iconSizes.tiny} style={styles.icon} icon={icon} />
      <Text style={styles.title}>{title}</Text>
    </View>
  );
};

// styles
const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.primary,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'center',
    borderRadius: borderRadius.extraLarge,
    paddingHorizontal: spacing.horizontal.small,
    paddingVertical: spacing.vertical.tiny,
    marginVertical: spacing.vertical.small,
  },
  icon: {
    tintColor: colors.primaryButton,
  },
  title: {marginLeft: spacing.horizontal.tiny, color: colors.text},
});
