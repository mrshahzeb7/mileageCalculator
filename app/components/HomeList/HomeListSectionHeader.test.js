import 'react-native';
import React from 'react';
import {HomeListSectionHeader} from './HomeListSectionHeader';
import renderer from 'react-test-renderer';

describe('HomeListSectionHeader', () => {
  it(' HomeListSectionHeader renders correctly', () => {
    const tree = renderer.create(<HomeListSectionHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
