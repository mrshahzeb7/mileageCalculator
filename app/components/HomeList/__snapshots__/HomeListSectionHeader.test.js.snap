// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeListSectionHeader  HomeListSectionHeader renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#2e2f32",
      "borderRadius": 76.48983229452425,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 26.68,
      "paddingHorizontal": 15,
      "paddingVertical": 13.34,
    }
  }
>
  <View>
    <Image
      style={
        [
          {
            "resizeMode": "contain",
          },
          null,
          {
            "height": 30.13699392404255,
            "width": 30.13699392404255,
          },
          {
            "tintColor": "#93b2fd",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "color": "#ffffff",
        "marginLeft": 7.5,
      }
    }
  />
</View>
`;
